src/nemo_evaluator/adapters/interceptors/turn_counter.py

```
@@ -95,18 +108,36 @@ def __init__(
         self._position = InjectionPosition(position)
         self._trigger = InjectionTrigger(trigger)
         self._interval = interval
+        self._pre_threshold_tool_reminder_interval = pre_threshold_tool_reminder_interval or None
         self._sessions: dict[str, _Session] = {}
         self._lock = asyncio.Lock()
         self._last_gc = time.monotonic()
 
+        if self._position is InjectionPosition.TOOL_MESSAGE and self._trigger is not InjectionTrigger.PERIODIC:
+            # Threshold reminders are WARN/URGENT escalation messages; keep
+            # tool-message placement limited to flat periodic reminders.
+            raise ValueError("position=tool_message is only supported with trigger=periodic")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)